src/ArborX_CrsGraphWrapper.hpp

```diff
@@ -41,6 +42,16 @@ inline void query(Tree const &tree, ExecutionSpace const &space,
   Kokkos::Profiling::popRegion();
 }
 
+template <typename Tree, typename Predicate, typename Callback>
+KOKKOS_FUNCTION void kernel_query(Tree const &tree, Predicate const &predicate,
+                                  Callback const &callback)
+{
```

Comment:
Honestly, I'm not sure whether this is the right interface. The alternative would be something like
```c++
struct BoundingVolumeHierarchy {
  ...
  template<typename Predicate, typename Callback>
  auto query(Predicate const& predicate, Callback const& callback);
```
This has an advantage of maintaining close interface to what we have right now.

Reply:
This is what I had first but I thought we prefer non-class calls. It's easy to add so that the non-class version would call this one.

Reply:
> This is what I had first but I thought we prefer non-class calls.

Don't know if that's true. There's no preference in my head, and I don't think we ever had a discussion.